gnuradio-runtime/swig/pmt_swig.i

```diff
@@ -302,7 +319,21 @@ namespace pmt{
   bool serialize(pmt_t obj, std::streambuf &sink);
   pmt_t deserialize(std::streambuf &source);
   void dump_sizeof();
-  std::string serialize_str(pmt_t obj);
-  pmt_t deserialize_str(std::string str);
+
+  std::vector<uint8_t> _serialize_str_u8(pmt_t obj);
+  pmt_t _deserialize_str_u8(std::vector<uint8_t> py_str);
 
 } //namespace pmt
+
+%pythoncode %{
+  def serialize_str(pmt_obj):
+    import array
+    tup = _serialize_str_u8(pmt_obj)
+    return array.array('B', tup).tobytes()
+
+  def deserialize_str(pmt_str):
+    tup = tuple([x for x in pmt_str])
```

Comment:
it should be possible to drop `[]` and do the list comprehension inside `tuple()`

Reply:
I don't get it. There is a list comprehension inside `tuple()`.

Reply:
But you should be able to do this `tuple(x for x in pmt_str)` as well (:

Reply:
Oh you mean *don't* do a list comprehension :grin:

Yeah you're right; in Py3k it's almost muscle memory to use the [] to force unrolling of iterators, but that `tuple()` will unroll the iterator as well.

Reply:
Ok, sorry, I meant generator expressions https://www.python.org/dev/peps/pep-0289/